packages/webpack/src/runtime/runtimeBuilder.js

```diff
@@ -21,6 +21,23 @@ module.exports = {
     }
 
     return {
+      /**
+       * Generates the preamble that caches selected globals to protect runtime
+       * from scuttling. The set of cached globals is limited to ones known to
+       * be used by runtime modules for the sake of bundle size.
+       *
+       * @returns {string}
+       */
+      getDefensiveCodingPreamble() {
+        const globals = [
+          'location', // AutoPublicPathRuntimeModule.js
+          'setTimeout', // LoadScriptRuntimeModule.js
+          'clearTimeout', // LoadScriptRuntimeModule.js
+          'document', // LoadScriptRuntimeModule.js, AutoPublicPathRuntimeModule.js
+          'self',
```

Comment:
just realised i _think_ [trustedTypes](https://github.com/webpack/webpack/blob/main/lib/runtime/GetTrustedTypesPolicyRuntimeModule.js#L31) needs to be here as well

```suggestion
          'trustedTypes', // GetTrustedTypesPolicyRuntimeModule.js
          'self',
```

Reply:
That's very niche, but maybe not in our target audience